### PR TITLE
chore: Enable the Source Process Latency metrics for all Sources

### PR DESCRIPTION
**Pull request by sarabala1979**
Signed-off-by: Saravanan Balasubramanian <sarabala1979@gmail.com>

### CODE REVIEW

**Comment by sarabala1979** [comment]
This is `show`  type

File: docs/METRICS.md [hunk 1]
```diff
@@ -48,6 +48,10 @@ Use this metric to determine how many retries performed for message processing.
 
 Golden metric type: error.
 
+### source_processLatencySeconds
```
Comment (alexec) [inline docs/METRICS.md:51]: should names use `_` rather than camel-case?

File: runner/sidecar/source/source.go [hunk 2]
```diff
@@ -3,13 +3,14 @@ package source
...
 	"io"
+	"time"
 )
 
 type Interface interface {
 	io.Closer
 }
 
-type Process func(ctx context.Context, msg []byte) error
+type Process func(ctx context.Context, msg []byte, ts time.Time) error
```
Comment (alexec) [inline runner/sidecar/source/source.go:13]: I don't think you need to change this signature. You can extract time from context using `MetaFromContext`.

File: runner/sidecar/sources.go [hunk 3]
```diff
@@ -78,11 +84,12 @@ func connectSources(ctx context.Context, process func(context.Context, []byte) e
 			return fmt.Errorf("duplicate source named %q", sourceName)
 		}
 
-		processWithRetry := func(ctx context.Context, msg []byte) error {
+		processWithRetry := func(ctx context.Context, msg []byte, ts time.Time) error {
 			span, ctx := opentracing.StartSpanFromContext(ctx, "processWithRetry")
 			defer span.Finish()
 			totalCounter.WithLabelValues(sourceName, fmt.Sprint(replica)).Inc()
 			totalBytesCounter.WithLabelValues(sourceName, fmt.Sprint(replica)).Add(float64(len(msg)))
+			processLatencyHistoGram.WithLabelValues(sourceName, fmt.Sprint(replica)).Observe(time.Now().UTC().Sub(ts).Seconds())
```
Comment (alexec) [inline runner/sidecar/sources.go:92]: nice

**Review by alexec — COMMENTED** [review]
any chance you can update `metrics_test.go` to check for this metric?

File: runner/sidecar/sources.go [hunk 4]
```diff
@@ -83,6 +89,12 @@ func connectSources(ctx context.Context, process func(context.Context, []byte) e
 			defer span.Finish()
 			totalCounter.WithLabelValues(sourceName, fmt.Sprint(replica)).Inc()
 			totalBytesCounter.WithLabelValues(sourceName, fmt.Sprint(replica)).Add(float64(len(msg)))
+			t, ok := ctx.Value(dfv1.MetaTime).(int64)
```
Comment (alexec) [inline runner/sidecar/sources.go:92]: please disallow message without meta-data,  use `dfv1.GetMetaFromContext` to get this

**Comment by sarabala1979** [comment]
> any chance you can update `metrics_test.go` to check for this metric?
Time being metrics show empty string. Can be added later.

**Review by alexec — APPROVED** [review]

**Pull request merged**